src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/EventResponse.java

```diff
@@ -0,0 +1,146 @@
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.servlet.http.HttpServletRequest;
+
+@Entity
+public class EventResponse {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    Long eventResponseId;
+
+    //The channel ID
+    String channelId;
+
+    //It identifies the watched resource, in our case the event, ID of the event
+    String resourceId;
+
+    //  example:  "https://www.googleapis.com/calendar/v3/calendars/my_calendar@gmail.com/events"
+    String resourceUri;
+
+    String resourceState;
+
+    Integer messageNumber;
+
+    @JsonInclude(Include.NON_NULL)
+    String channelExpiration;
+
+    @JsonInclude(Include.NON_NULL)
+    String channelToken;
+
+    public EventResponse() {
+    }
+
+    public EventResponse(HttpServletRequest request) {
+        this.channelId = request.getHeader("X-Goog-Channel-ID");
+        this.resourceId = request.getHeader("X-Goog-Resource-ID");
+        this.resourceUri = request.getHeader("X-Goog-Resource-URI");
+        this.resourceState = request.getHeader("X-Goog-Resource-State");
+        this.messageNumber = Integer.parseInt(request.getHeader("X-Goog-Message-Number"));
+        this.channelExpiration = request.getHeader("X-Goog-Channel-Expiration");
+        this.channelToken = request.getHeader("X-Goog-Channel-Token");
+    }
+
+    public Long getEventResponseId() {
+        return eventResponseId;
+    }
+
+    public void setEventResponseId(Long eventResponseId) {
+        this.eventResponseId = eventResponseId;
+    }
+
+    public String getChannelId() {
+        return channelId;
+    }
+
+    public void setChannelId(String channelId) {
+        this.channelId = channelId;
+    }
+
+    public String getResourceId() {
+        return resourceId;
+    }
+
+    public void setResourceId(String resourceId) {
+        this.resourceId = resourceId;
+    }
+
+    public String getResourceUri() {
+        return resourceUri;
+    }
+
+    public void setResourceUri(String resourceUri) {
+        this.resourceUri = resourceUri;
+    }
+
+    public String getResourceState() {
+        return resourceState;
+    }
+
+    public void setResourceState(String resourceState) {
+        this.resourceState = resourceState;
+    }
+
+    public Integer getMessageNumber() {
+        return messageNumber;
+    }
+
+    public void setMessageNumber(Integer messageNumber) {
+        this.messageNumber = messageNumber;
+    }
+
+    public String getChannelExpiration() {
+        return channelExpiration;
+    }
+
+    public void setChannelExpiration(String channelExpiration) {
+        this.channelExpiration = channelExpiration;
+    }
+
+    public String getChannelToken() {
+        return channelToken;
+    }
+
+    public void setChannelToken(String channelToken) {
+        this.channelToken = channelToken;
+    }
+
+    public HttpStatus validate() {
```

Comment:
validate should decide if the model data is valid, the return type should be boolean, then in the controller you can either do this or that